src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,633 @@
+            )),
+            _ => Ok(description),
+        }
+    }
+}
+
+impl<C: Context> Parse<C> for simple_extensions::ArgumentsItem {
+    type Parsed = ArgumentsItem;
+    type Error = ArgumentsItemError;
+
+    fn parse(self, ctx: &mut C) -> Result<Self::Parsed, Self::Error> {
+        match self {
+            simple_extensions::ArgumentsItem::EnumerationArg(arg) => Ok(ctx.parse(arg)?.into()),
+            simple_extensions::ArgumentsItem::ValueArg(arg) => Ok(ctx.parse(arg)?.into()),
+            simple_extensions::ArgumentsItem::TypeArg(arg) => Ok(ctx.parse(arg)?.into()),
+        }
+    }
+}
+
+impl From<ArgumentsItem> for simple_extensions::ArgumentsItem {
+    fn from(value: ArgumentsItem) -> Self {
+        match value {
+            ArgumentsItem::EnumArgument(arg) => arg.into(),
+            ArgumentsItem::ValueArgument(arg) => arg.into(),
+            ArgumentsItem::TypeArgument(arg) => arg.into(),
+        }
+    }
+}
+
+/// Parse errors for [simple_extensions::ArgumentsItem].
+#[derive(Debug, Error, PartialEq)]
+pub enum ArgumentsItemError {
+    /// Invalid enumeration options.
+    #[error("invalid enumeration options: {0}")]
+    InvalidEnumOptions(#[from] EnumOptionsError),
+
+    /// Empty optional field.
+    #[error("the optional field `{0}` is empty and should be removed")]
+    EmptyOptionalField(String),
+}
+
+/// Arguments that support a fixed set of declared values as constant arguments.
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumerationArg {
+    /// A human-readable name for this argument to help clarify use.
+    name: Option<String>,
+
+    /// Additional description for this argument.
+    description: Option<String>,
+
+    /// List of valid string options for this argument.
+    options: EnumOptions,
+}
+
+impl<C: Context> Parse<C> for simple_extensions::EnumerationArg {
+    type Parsed = EnumerationArg;
+
+    type Error = ArgumentsItemError;
+
+    fn parse(self, ctx: &mut C) -> Result<EnumerationArg, ArgumentsItemError> {
+        Ok(EnumerationArg {
+            name: ArgumentsItem::parse_name(self.name)?,
+            description: ArgumentsItem::parse_description(self.description)?,
+            options: ctx.parse(self.options)?,
+        })
+    }
+}
+
+impl From<EnumerationArg> for simple_extensions::EnumerationArg {
+    fn from(value: EnumerationArg) -> Self {
+        simple_extensions::EnumerationArg {
+            name: value.name,
+            description: value.description,
+            options: value.options.into(),
+        }
+    }
+}
+
+impl From<EnumerationArg> for simple_extensions::ArgumentsItem {
+    fn from(value: EnumerationArg) -> Self {
+        simple_extensions::ArgumentsItem::EnumerationArg(value.into())
+    }
+}
+
+impl From<EnumerationArg> for ArgumentsItem {
+    fn from(value: EnumerationArg) -> Self {
+        ArgumentsItem::EnumArgument(value)
+    }
+}
+
+/// List of valid string options
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumOptions(Vec<String>);
```

Comment:
Should we switch this to store the `HashSet` that we build in the `Parse` impl? We lose the order, but that doesn't change the meaning. It captures that these options are unique.

Reply:
Sounds good. Fixed in [commit](https://github.com/substrait-io/substrait-rs/pull/185/commits/c16c017dce3ade6928194f65b6b30c01ce252bb2).